sklearn/datasets/openml.py

```diff
@@ -0,0 +1,157 @@
+import json
+import numbers
+import sys
+import os
+from os.path import join, exists
+from warnings import warn
+
+try:
+    # Python 2
+    from urllib2 import urlopen
+except ImportError:
+    # Python 3+
+    from urllib.request import urlopen
+    # from http.client import IncompleteRead
+
+from scipy.io.arff import loadarff
+import numpy as np
+
+from .base import get_data_home
+from ..externals.joblib import Memory
+from ..externals.six import StringIO
+from ..utils import Bunch
+
+_SEARCH_NAME = "https://openml.org/api/v1/json/data/list/data_name/{}/limit/1"
+_DATA_INFO = "https://openml.org/api/v1/json/data/{}"
+
+
+def _get_data_info_by_name(name, version):
+    if version == "active":
+        json_string = urlopen(_SEARCH_NAME.format(name + "/status/active/"))
+    else:
+        json_string = urlopen(_SEARCH_NAME.format(name)
+                              + "/data_version/{}".format(version))
+    json_data = json.load(json_string)
+    return json_data['data']['dataset'][0]
```

Comment:
For non-existent datasets, all the user will see is `KeyError: 'data'`. Need more informative error message for inactive and absent.